elrond-wasm-derive/src/generate/abi_gen.rs

```diff
@@ -89,14 +87,28 @@ fn generate_abi_method_body(
                     #endpoint_def
                     contract_abi.endpoints.push(endpoint_abi);
                 })
-            } else {
-                None
-            }
+            },
+            _ => None,
         })
-        .collect();
+        .collect()
+}
 
-    let supertrait_snippets: Vec<proc_macro2::TokenStream> = if is_contract_main {
-        contract
+fn has_callback(contract: &ContractTrait) -> bool {
+    let callback_count = contract
+        .methods
+        .iter()
+        .filter(|m| {
+            matches!(
+                m.public_role,
+                PublicRole::Callback(_) | PublicRole::CallbackRaw
+            )
+        })
+        .count();
+    callback_count > 0
+}
```

Comment:
```suggestion
fn has_callback(contract: &ContractTrait) -> bool {
    contract
        .methods
        .iter()
        .any(|m| {
            matches!(
                m.public_role,
                PublicRole::Callback(_) | PublicRole::CallbackRaw
            )
        })
}
```
Use `any` instead of `filter` + `count() > 0`.